bundle/config/root.go

```diff
@@ -338,13 +338,23 @@ func (r *Root) MergeTargetOverrides(name string) error {
 		"resources",
 		"sync",
 		"permissions",
-		"variables",
 	} {
 		if root, err = mergeField(root, target, f); err != nil {
 			return err
 		}
 	}
 
+	// Merge `variables`. This field must be overwritten if set, not merged.
+	if v := target.Get("variables"); v.Kind() != dyn.KindInvalid {
+		_, err = dyn.Map(v, ".", dyn.Foreach(func(p dyn.Path, variable dyn.Value) (dyn.Value, error) {
+			root, err = dyn.SetByPath(root, dyn.MustPathFromString("variables").Append(p...), variable)
+			return root, err
+		}))
```

Comment:
This overwrites everything, including the type and description. I believe this would allow changing the type (complex -> simple or vice versa) or adding/changing descriptions, which is contrary to what we would want.

Note that this is probably already the existing behavior so can be dealt with separately.

Reply:
Addressed in the latest commit